core/markdown_api.php

```diff
@@ -0,0 +1,106 @@
+<?php
+# MantisBT - A PHP based bugtracking system
+
+# MantisBT is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# MantisBT is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with MantisBT.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Markdown API
+ *
+ * @package CoreAPI
+ * @subpackage MarkdownAPI
```

Comment:
I also agree of removing the api, it looks an old school approach. 

We could not make the methods static (MantisMarkdown::markdown_text or MantisMarkdown::text()) and so on...as its parent library is non-static. 

I suggest to make the api as a service container instead:

```
class MantisMarkdownApi {
    
    function __construct(){
        load the MantisMarkdown class
    }
    static function text($text){
        return $MantisMarkdown->text($text);    
    }
    
    static function line($text){
        return $MantisMarkdown->line($text);    
    }
}
```

Let me know your thoughts guys?